_script/deploy

```diff
@@ -3,6 +3,39 @@
 set -exuo pipefail
 cd $(dirname $0)/..
 
+function docker_authenticate() {
+    DOCKER_USERNAME=${DOCKER_USERNAME:-""}
+    DOCKER_PASSWORD=${DOCKER_PASSWORD:-""}
+    if $(test "$DOCKER_USERNAME" != "" ) && $(test "$DOCKER_PASSWORD" != "")
+    then
+        echo "INFO: Docker env variables set. Logging in"
+        docker login -u "$DOCKER_USERNAME" -p "$DOCKER_PASSWORD"
+        return $?
+    fi
+    local_docker_config_store="${HOME}/.docker"
+    if $(test -f "${local_docker_config_store}/config.json") && $(test -r "${local_docker_config_store}/config.json")
```

Comment:
does "test -r" include "test -f" scope? I mean, just using -r, is it not enough?

Reply:
You are right... i think, at some stage, i was differentiating to provide more specific error messages, but its not worth (both checks are there just because of uncontrolled inheritance xD"